downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/ProtegeExporter.java

```diff
@@ -0,0 +1,414 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+/**
+ * Exports top-level pathways in Protege format, using the Perl code in GKB::WebUtils, and then
+ * packages the resulting files into a single tar file.
+ * Takes ~1.5 hours on my (Solomon) workstation, with 7 threads running 8 cores. Of course, it ran the system
+ * at a very high load. 5 threads (concurrent protege exporters) finished in almost the same amount of time
+ * and the system was not too overloaded. Less than that seemed to complete slower.
+ * @author sshorser
+ *
+ */
+public class ProtegeExporter
+{
+	// The code in GKB::WebUtils always writes protege files to /tmp/ and it doesn't look like that is configurable,
+	// so we'll work in /tmp as well.
+	private static final String PROTEGE_ARCHIVE_PATH = "/tmp/protege_files.tar";
+	private static final String PROTEGE_FILES_DIR = PROTEGE_ARCHIVE_PATH.replace(".tar", "/");
+	private static final Logger logger = LogManager.getLogger();
+	private String releaseDirectory;
+	private int parallelism = ForkJoinPool.getCommonPoolParallelism();
+	private List<String> extraIncludes = new ArrayList<>();
+	private String pathToWrapperScript = "";
+	private String downloadDirectory;
+	private Set<Long> pathwayIdsToProcess = new HashSet<>();
+	private Set<String> speciesToProcess = new HashSet<>();
+	
+	public ProtegeExporter()
+	{
+		// Default no-arg consturctor.
+	}
+	
+	public ProtegeExporter(Properties props, String releaseDir, String downloadDir)
+	{
+		this.setReleaseDirectory(releaseDir);
+		String propsPrefix = "protegeexporter";
+		// Use the value of CommonPoolParallelism if nothing is in the properties file.
+		int degP = Integer.parseInt(props.getProperty(propsPrefix + ".parallelism", String.valueOf(ForkJoinPool.getCommonPoolParallelism())));
+		this.setParallelism(degP);
+		String extraIncludeStr = props.getProperty(propsPrefix+".extraIncludes");
+		if (extraIncludeStr != null && !extraIncludeStr.trim().isEmpty())
+		{
+			List<String> extraIncs = Arrays.asList(extraIncludeStr.split(","));
+			this.setExtraIncludes(extraIncs);
+		}
+		String pathToWrapper = props.getProperty(propsPrefix+".pathToWrapperScript");
```

Comment:
Comment for each of the props? I know they are explained in the readme, but I like small explainer comments in the code

Reply:
I have written detailed documentation for the setter methods. I don't want to re-document the same thing too many times in multiple places, because if something changes then there are more areas of documentation that need to be kept in sync.

Reply:
You could add a comment "See 'method name'" if it's already documented elsewhere.